feature: Add groups created by addPbxGroup() to the meta PBXGroup

tdermendjiev opened the pull request:
Current implementation doesn't add created groups to the main/meta PBXGroup and therefore they don't show up in the file tree.

Fatme commented on lib/pbxProject.js:
```diff
@@ -1,11 +1,12 @@
     f = util.format,
     EventEmitter = require('events').EventEmitter,
-    path = require('path'),
+    $path = require('path'),
```
I'm wondering why we are using `$` before path.

tdermendjiev replied:
We need to differentiate between the global (used for path module) and local (passed in most of the methods) variable.

Fatme commented on lib/pbxProject.js:
```diff
@@ -286,9 +287,24 @@ pbxProject.prototype.removeFromPbxBuildFileSection = function (file) {
+pbxProject.prototype.addPbxGroup = function (filePathsArray, name, path, sourceTree, opt) {
     var groups = this.hash.project.objects['PBXGroup'],
-        pbxGroupUuid = this.generateUuid(),
+        pbxGroupUuid = opt.uuid ? opt.uuid : this.generateUuid(),
```
`opt.uuid || this.generateUuid()` is shorter

Fatme commented on lib/pbxProject.js:
```diff
@@ -286,9 +287,24 @@ pbxProject.prototype.removeFromPbxBuildFileSection = function (file) {
+    var groups = this.hash.project.objects['PBXGroup'];
+    var candidates = [];
+    for (var key in groups) {
+        if (groups[key].path == undefined && groups[key].name == undefined && groups[key].isa) {
```
Maybe this is better
```
if (groups[key] && !groups[key].path && !groups[key].name && groups[key].isa) { .. }
```

Fatme commented on lib/pbxProject.js:
```diff
@@ -286,9 +287,24 @@ pbxProject.prototype.removeFromPbxBuildFileSection = function (file) {
+        return candidates[0];
+    }
+
+    return null;
```
In case when we have more than one candidate, shouldn't return the first one? Actually is it possible to have more than one candidate?

tdermendjiev replied:
We should have exactly one candidate as this will be the root group. If they are more than one there should be an error. Maybe we can throw an exception.

tdermendjiev committed:
feat: Add support various file extensions and export filetypes for pbxFile module

tdermendjiev committed:
feat: Implement adding and removing of pbx groups

*support for c++ files added
*recursively add pbxGroup and its children
*recursively remove pbxGroup and its children
*make sure pbx group name is not duplicated on adding new group

Fatme commented on lib/pbxFile.js:
```diff
@@ -11,13 +15,42 @@ var path = require('path'),
 
+function fileTypes() {
+    return {
+		SOURCE_FILE,
```
Identation issue

Fatme commented on lib/pbxProject.js:
```diff
@@ -320,23 +344,80 @@ pbxProject.prototype.addPbxGroup = function (filePathsArray, name, path, sourceT
 }
 
+pbxProject.prototype.removePbxGroup = function(name, path) {
+        var group = this.pbxGroupByName(name);
```
Identation problem

Fatme reviewed (approved):
After addressing identation problem

tdermendjiev committed:
feat: add support for header search path deletion by string
